src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -3344,11 +3344,7 @@ void MacroAssembler::check_klass_subtype_slow_path(Register sub_klass,
   mv(x10, super_klass);
 
 #ifndef PRODUCT
-  mv(t1, (address)&SharedRuntime::_partial_subtype_ctr);
-  Address pst_counter_addr(t1);
-  ld(t0, pst_counter_addr);
-  add(t0, t0, 1);
-  sd(t0, pst_counter_addr);
+  incrementw(ExternalAddress((address)&SharedRuntime::_partial_subtype_ctr));
```

Comment:
I just checked the x86 and aarch64 counterpart. Seems that aarch64 bears the same issue [1] as it uses `ldr` & `str` which load and store 64-bit data items like here.

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp#L1565

Reply:
I'll open a separate PR, and maybe fix it.

Reply:
FYI: https://github.com/openjdk/jdk/pull/19011